common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		this.drawGuiScroll();
+		if (prevPage.visible && nextPage.visible) this.drawCenteredArrowUnderlay(area$offsetCenterX * 2);
+		this.drawTitle();
+	}
+
+	@Override
+	public void initGui() {
+		this.tabList.clear();
+		this.getDataFromNBT();
+		this.calculateConstraints();
+		this.updateTitle();
+		this.slider = this.addButton(new GuiGuideBookSlider(0, gui$sliderX, gui$sliderY0, gui$sliderY1, z$titleButtons, currentPage.getProgress(), gui$sliderWidth, gui$sliderHeight));
+		this.slider.enabled = false;
+		this.home = this.addButton(new GuiGuideBookTexturedButton(1, gui$x3 - gui$tagWidth, gui$y0 - gui$tagHeight, gui$tagWidth, gui$fullTagHeight, 16, 64, z$titleButtons, 128, 0, 16, 16, false, GuiGuideBookTexturedButton.EnumButtonType.TAB));
+		this.prevPage = this.addButton(new GuiGuideBookTexturedButton(2, area$xCenter - area$offsetCenterX, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 144, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		this.nextPage = this.addButton(new GuiGuideBookTexturedButton(3, area$xCenter + area$offsetCenterX - gui$arrowWidth, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 168, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		for (int i = 0; i < maxTabs; i++) {
+			tabList.add(new GuiGuideBookTabButton(i, gui$x3 - 2 - 2 * gui$tagWidth - (i * gui$tagWidth), gui$y0, gui$tagWidth, gui$tagHeight, ""));
+		}
+		this.updateButtonVisibility();
+	}
+
+	@Override
+	public boolean doesGuiPauseGame() {
+		return false;
+	}
+
+	@Override
+	public void onGuiClosed() {
+		currentPage.setProgress(slider.getProgress());
+		ItemGuideBook.setCurrentPage(currentPage, hand);
+		super.onGuiClosed();
+	}
+
+	@Override
+	protected void actionPerformed(GuiButton button) throws IOException {
+		switch (button.id) {
+			case 1:
+				state = menu;
+				slider.reset();
+				break;
+			case 2:
+				prevPage();
+				break;
+			case 3:
+				nextPage();
+				break;
+			default:
+				break;
+		}
+		updateTitle();
+		updateButtonVisibility();
+	}
+
+	@Override
+	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
+		if (state == menu)
+			for (MenuItemsDivision div : divisionsList) {
+				for (MenuItem item : div.getList()) {
+					if (item.mousePressed()) {
+						item.playPressSound(this.mc.getSoundHandler());
+						this.pressedItem(item);
+						updateButtonVisibility();
+						return;
+					}
+				}
+			}
+		for (GuiGuideBookTabButton tab : tabList) {
+			if (tab.mousePressed(mc, mouseX, mouseY)) {
+				tab.playPressSound(this.mc.getSoundHandler());
+				if (!tab.isSet && state != menu) {
+					PageInformation setPage = new PageInformation();
+					setPage.page = currentPage.page;
+					setPage.progress = currentPage.progress;
+					if (savedTabs.addTab(new SavedTab(setPage, state))) {
+						tab.isSet = true;
+						if (tab.isSet) tab.isActive = true;
+					}else{
```

Comment:
Give the poor *else* some space, duh

Reply:
I need to `Ctrl+Alt+L` more often I know I now x)

Reply:
I pretty much press that every time I stop typing. There's also a "format on commit" option in IntelliJ's commit dialog, but that will format every file you've touched, even the parts of it that you didn't touch, so it's not a good idea in this project. I do need to make a "format everything" commit at some point though, so that that won't be a problem anymore.

Reply:
I have to develop the habit